src/trix/models/composition.coffee

```diff
@@ -478,6 +493,15 @@ class Trix.Composition extends Trix.BasicObject
   notifyDelegateOfInsertionAtRange: (range) ->
     @delegate?.compositionDidPerformInsertionAtRange?(range)
 
+  getPositionInBlock: ->
+    [startPosition, endPosition] = @getSelectedRange()
+    startLocation = @document.locationFromPosition(startPosition)
+    endLocation = @document.locationFromPosition(endPosition)
+    block = @document.getBlockAtIndex(endLocation.index)
+    return "end" if @getBlock()?.getConfig("breakOnReturn") and block.text.getStringAtRange([endLocation.offset, endLocation.offset + 1]) is "\n"
+    return "start" if @getBlock()?.getConfig("breakOnReturn") and block.text.getStringAtRange([endLocation.offset - 1, endLocation.offset]) is ""
+    return [startPosition, endPosition] if @getBlock()?.getConfig("breakOnReturn")
```

Comment:
I think this method would be easier to read as a traditional if/else if/else expression rather than a bunch of conditioned returns.